core/src/main/java/org/elasticsearch/common/Priority.java

```diff
@@ -16,28 +16,29 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+
 package org.elasticsearch.common;
 
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 
 import java.io.IOException;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
 
-/**
- *
- */
-public final class Priority implements Comparable<Priority> {
+public enum Priority {
+
+    IMMEDIATE((byte) -1),
```

Comment:
now that we do a full cluster restart shall we fix this and start with `0` again?